### PR TITLE
[DX-1774, TT-13333] Upstream

### PR DESCRIPTION
**Pull request by andyo-tyk**
### Preview Link
https://deploy-preview-5746--tyk-docs.netlify.app/docs/nightly/basic-config-and-security/security/upstream-authentication/

### Description
Upstream authentication

___

### **PR Type**
Documentation

___

### **Description**
- Added a new documentation page on upstream authentication, detailing various methods such as Mutual TLS, Token-based authentication, Basic Authentication, and OAuth 2.0.
- Provided configuration examples for setting up upstream authentication in Tyk OAS API definitions.
- Updated the documentation menu to include a new entry for "Upstream Authentication" and made title adjustments for consistency.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>upstream-authentication.md</strong><dd><code>Add detailed upstream authentication documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/basic-config-and-security/security/upstream-authentication.md

<li>Added comprehensive documentation on upstream authentication methods.<br> <li> Included sections on Mutual TLS, Token-based authentication, Basic <br>Authentication, and OAuth 2.0.<br> <li> Provided configuration examples for Tyk OAS API definitions.<br> <li> Explained usage of Tyk API Designer for setting up upstream <br>authentication.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5746/files#diff-5e8ca84fb917fa0edcb561edfca2963da5b571176dc00c4dfe6e74702878f1ee">+437/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>menu.yaml</strong><dd><code>Update menu structure for upstream authentication</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/data/menu.yaml

<li>Updated menu titles for consistency (e.g., "MTLS" to "Mutual TLS").<br> <li> Added new menu entry for "Upstream Authentication".<br> <li> Adjusted capitalization for "Authentication and Authorization".<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5746/files#diff-a6789c66bfa0660f7fdbe4796002ab4777677bae59a0276689c840100ff0b0b2">+11/-3</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: tyk-docs/content/basic-config-and-security/security/upstream-authentication.md [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+    - OAuth
+    - mTLS
+    - Basic Auth
+description: Authenticating Tyk Gateway with upstream services
+date: "2024-11-18"
+---
+
+Tyk Gateway sits between your clients and your services, securely routing requests and responses. For each API proxy that you expose on Tyk, you can configure a range of different methods that clients must use to identify (authenticate) themselves to Tyk Gateway. These are described in detail in the [Client Authentication]({{< ref "basic-config-and-security/security/authentication-&-authorization" >}}) section.
+
+In the same way as you use Client Authentication to securely confirm the identity of the API clients, your upstream services probably need to securely confirm the identity of their client - namely Tyk. This is where Tyk's flexible **Upstream Authentication** capability comes in.
```
Comment (sharadregoti) [inline tyk-docs/content/basic-config-and-security/security/upstream-authentication.md]: Need an image that elaborate this text.

File: tyk-docs/content/basic-config-and-security/security/upstream-authentication.md [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+
+<hr>
+
+## Token-based authentication
+
+Token-based authentication (also referred to as Auth Token) is a method whereby the client is identified and authenticated by the server based on a key/token they present as a credential with each request. Typically the token is issued by the server to a specific client.
+
+The server determines how the key should be provided - typically in a request header, cookie or query parameter.
+
+Tyk supports Auth Token as a method for authenticating **clients** with the **Gateway** - you can use Tyk Gateway or Dashboard to generate access *keys* for an Auth Token protected API as explained in the [documentation]{{{< ref "" >}}}. The client must then provide the *key* in the appropriate parameter for each request.
```
Comment (sharadregoti) [inline tyk-docs/content/basic-config-and-security/security/upstream-authentication.md]:
```suggestion
Tyk supports [Auth Token]({{< ref "api-management/authentication-authorization/#use-bearer-tokens" >}}) as a method for authenticating **clients** with the **Gateway** - you can use Tyk Gateway or Dashboard to generate access *keys* for an Auth Token protected API as explained in the [documentation]({{< ref "api-management/authentication-authorization/#enable-bearer-tokens-in-your-api-definition-with-the-dashboard" >}}). The client must then provide the *key* in the appropriate parameter for each request.
```

File: tyk-docs/content/basic-config-and-security/security/upstream-authentication.md [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+
+## Basic Authentication
+
+Basic Authentication is a standard authentication mechanism implemented by HTTP servers, clients and web browsers. This makes it an excellent access control method for smaller APIs.
+
+An API request made using Basic Authentication will have an `Authorization` header that contains the client's credentials in the form: `Basic <credentials>`.
+
+The `<credentials>` are a base64 encoded concatenation of a client username and password, joined by a single colon `:`.
+
+Tyk supports Basic Authentication as a method for authenticating **clients** with the **Gateway** - you can use Tyk Gateway or Dashboard to create Basic Auth users, as explained in the [documentation]{{{< ref "basic-config-and-security/security/authentication-authorization/basic-auth#protect-your-api-with-basic-authentication" >}}}.
```
Comment (sharadregoti) [inline tyk-docs/content/basic-config-and-security/security/upstream-authentication.md]:
```suggestion
Tyk supports Basic Authentication as a method for authenticating **clients** with the **Gateway** - you can use Tyk Gateway or Dashboard to create Basic Auth users, as explained in the [documentation]({{< ref "basic-config-and-security/security/authentication-authorization/basic-auth#protect-your-api-with-basic-authentication" >}}).
```

File: tyk-docs/content/basic-config-and-security/security/upstream-authentication.md [hunk 4]
```diff
@@ -0,0 +1,437 @@
...
+
+The Authentication Server (auth server) has the concept of an OAuth Client - this is equivalent to the client's account on the auth server. There are different ways that a client can authenticate with the auth server, each with their own advantages and disadvantages for different use cases.
+
+The auth server is often managed by a trusted third party Identity Provider (IdP) such as Okta or Auth0.
+
+Tyk supports OAuth 2.0 as a method for authenticating **clients** with the **Gateway** - you can use Tyk's own auth server functionality via the [Tyk OAuth 2.0]{{{< ref "basic-config-and-security/security/authentication-authorization/oauth-2-0" >}}} auth method or obtain the access token via a third party auth server and use the [JWT Auth]({{< ref "basic-config-and-security/security/authentication-authorization/json-web-tokens" >}}) method.
```
Comment (sharadregoti) [inline tyk-docs/content/basic-config-and-security/security/upstream-authentication.md]:
```suggestion
Tyk supports OAuth 2.0 as a method for authenticating **clients** with the **Gateway** - you can use Tyk's own auth server functionality via the [Tyk OAuth 2.0]({{< ref "basic-config-and-security/security/authentication-authorization/oauth-2-0" >}}) auth method or obtain the access token via a third party auth server and use the [JWT Auth]({{< ref "basic-config-and-security/security/authentication-authorization/json-web-tokens" >}}) method.
```

**Review by sharadregoti — APPROVED** [review]

File: tyk-docs/content/basic-config-and-security/security/upstream-authentication.md (hunk 1 above)
Comment (andyo-tyk) [reply tyk-docs/content/basic-config-and-security/security/upstream-authentication.md]: I agree - there are a lot of places where we need images (and can then remove some of my verbosity 😂 ). Given workload at the moment, can we add this later, for when this page is published publicly?

Comment (sharadregoti) [reply tyk-docs/content/basic-config-and-security/security/upstream-authentication.md]: I'll try to add it by EOD, If not then I'll merge it.

**Review by lghiur — APPROVED** [review]

**Review by jeffy-mathew — APPROVED** [review]

**Pull request merged**